include/nbl/asset/filters/CBlitImageFilter.h

```diff
@@ -409,7 +418,63 @@ class CBlitImageFilter : public CImageFilter<CBlitImageFilter<Normalize,Clamp,Sw
 					if (otherScale)
 					for (auto k=0; k<MaxChannels; k++)
 						scale.factor[k] *= otherScale->factor[k];
-						
+
+					// compute the phaseSupportLUT
+
+					// a dummy load functor
+					// does nothing but fills up the `windowSample` with 1s (identity) so we can preserve the value of kernel
+					// weights when eventually `windowSample` gets multiplied by them later in
+					// `CFloatingPointSeparableImageFilterKernelBase<CRTP>::sample_functor_t<PreFilter,PostFilter>::operator()`
+					// this exists only because `evaluateImpl` expects a pre filtering step.
+					// 
+					// Here I suggest to make something like a CImageFilterKernel<CRTP,value_type>::computeWeights which goes through
+					// the same pipeline of injecting other pre filtering stages (like `wrap` for the `CScaledImageFilterKernel`, for example)
+					// and which doesn't take this functor as argument.
```

Comment:
there's an easier API/function for just evaluating a kernel
https://github.com/Devsh-Graphics-Programming/Nabla/blob/master/include/nbl/asset/filters/kernels/CommonImageFilterKernels.h#L83

did you not see it before or did it not fullfill some criteria?

Reply:
wanted to use the existing infrastructure of injecting other prefiltering stages
for example, in `CScaledImageFilterKernel` you have this https://github.com/achalpandeyy/Nabla/blob/master/include/nbl/asset/filters/kernels/CScaledImageFilterKernel.h#L93
where the input kernel evaluation point gets rescaled (potentially) multiple times to get a value in the defined domain of the kernel (for example [-1/2,1/2] for the Box kernel)

if I would have just called `weight` I would have to implement something like the above myself

Reply:
and thats just `CScaledImageFilterKernel` other kernels might inject their own pre and post filtering stages

Reply:
hmm ok I guess you're right, that you need this weirdness for it to work with scaling,

but it whoul all be moved to some static method to precompute the LUT

Reply:
Resolved.